### ISSUE TITLE
Add-SPOFieldToContentType throws The collection has not been initialized when trying to add Add-SPOTaxonomyField in same instance of PowerShell window as 

### ISSUE THREAD

**Issue by sanjeev006** [OfficeDev/PnP#971]
We are getting following error when trying to execute Add-SPOTaxonomyField and add that field using Add-SPOFieldToContentType on same instance of PowerShell window. Regular site columns works without problem.

If we execute Add-SPOTaxonomyField, close PowerShell ISE and open and execute Add-SPOFieldToContentType and add texonomy site columns the it works without any problem. Tried putting delay in between Add-SPOTaxonomyField and Add-SPOFieldToContentType and that didn't work either.

To reproduce this issue - try

Add-SPOTaxonomyField -DisplayName 'Department' -InternalName 'ABC_Department' .....
Add-SPOFieldToContentType -Field 'ABC_Department' -ContentType 'Test CT'

We are using June release of msi.

Thanks.

Error - 
format-default : The collection has not been initialized. It has not been requested or the request has not been executed. It may need to be explicitly requested.
    + CategoryInfo          : NotSpecified: (:) [format-default], CollectionNotInitializedException
    + FullyQualifiedErrorId : Microsoft.SharePoint.Client.CollectionNotInitializedException,Microsoft.PowerShell.Commands.FormatDefaultCommand

**Comment by erwinvanhunen** [comment]
Hi!

I just tried to reproduce this issue but I fail, e.g. it works as expected in my environment. Are you using SharePoint Online or on-premises?

/Erwin

**Comment by sanjeev006** [comment]
Hi Erwin,

I am deploying into SP O365 environment. If I split the code into two
files, open PowerShell IDE and create taxonomy field, close PowerShell
window and open new instance of PS window and run add field to CT command
then it works without any problem. If I try to create taxonomy field and
immediately try to add it to CT it fails. We tried with we other people in
my office and we all got same error.

This 2nd project where I am trying to use PnP commands to deploy to SP
O365. We had seen same issue with previous build around 2 months ago. Not
sure why it can't run in sequence and in same instance of PS IDE window.

Thanks

**Comment by sanjeev006** [comment]
Using Windows 8.1 Enterprise and Windows PowerShell ISE to execute commandlets.

**Comment by erwinvanhunen** [comment]
Have you tried it against different tenants?

**Comment by sanjeev006** [comment]
Yes. 3 different tenants and multiple site collections including CT Hub in two separate tenants.

**Comment by erwinvanhunen** [comment]
I actually think I have an idea what is causing this. When you check the content type, is the field actually there? I mean where you run both commands in one go.

The error message seems to be related to the actual return value of the Add-SPOTaxonomyField cmdlet. The moment the cmdlet is created, it returns the field object to the output. In the installation of the PowerShell cmdlets, there is a format file (called OfficeDevPnP.PowerShell.Commands.Format.ps1xml) that defines what properties of a field should be shown. If that format file cannot be found PowerShell will use the default way of rendering the object. It goes wrong there because some of the properties of the field have not been loaded (this is the default behavior of CSOM, to save bandwidth). That's the error message you're seeing: PowerShell tries to show a property value but that property value is not available yet.

This should as such not hinder your powershell script. A simple way to test it is by doing this:

$f = Add-SPOTaxonomyField -DisplayName 'Department' -InternalName 'ABC_Department' .....
Add-SPOFieldToContentType -Field 'ABC_Department' -ContentType 'Test CT'

(notice the $f = in the beginning). That way I store the output of the Add-SPOTaxonomyField cmdlet in a variable, but I don't have to do anything with it. The thing is that PowerShell will not try to render the field, and likely, in your case, you will not receive an error message.

Please double check in your installation if the OfficeDevPnP.PowerShell.Commands.Format.ps1xml is available in the same folder as OfficeDevPnP.PowerShell.Commands.dll as that file defines how certain objects are being rendered.

**Comment by iddycol** [comment]
Hi Erwin,

Try provisioning a taxonomy field, delete it from the UI, them provisioning it again. This can give a field already exists error because the taxonomy hidden field is hanging around. Don't know if that is related to this issue but I noticed it when playing with creating taxonomy site columns using the apply templates extension. There is code in pnp core to fix this but it's not wired into that flow.

Just ignore me if this isn't relevant!

Regards

Adrian

**Comment by sanjeev006** [comment]
Hi Erwin,

Thank you for prompt response. It turned out to be issue with Add-SPOContentType. I followed your example and added variable in front of Add-SPOContentType and it worked. No more error. I had to do same for all 8 content types that I am creating on our new project. 

$ctNews = Add-SPOContentType -Name $ctName.....

Now I can initialize PnP, add terms in term store, add taxonomy fields, add site columns and create content types and add field - all in single instance of PowerShell.

Since we are trying to use PnP PS script to provision all SP O365 projects - so I might ping you with few more questions/issues.

I didn't have to store Add-SPOTaxonomyField in temp variable as in $f = Add-SPOTaxonomyField.

Thanks,

Sanjeev
